util/memory/tracker.go

```diff
@@ -350,6 +375,15 @@ func (t *Tracker) Consume(bytes int64) {
 	}
 }
 
+// BufferedConsume is used to buffer memory usage and do late consume
+func (t *Tracker) BufferedConsume(bufferedMemSize *int64, bytes int64) {
+	*bufferedMemSize += bytes
+	if *bufferedMemSize > int64(config.TrackMemWhenExceeds) {
+		t.Consume(*bufferedMemSize)
+		*bufferedMemSize = int64(0)
+	}
+}
```

Comment:
Why we need this method?

Reply:
To reflect the memory usage asap, if for-loop runs for a long time.